### PR TITLE
PHPStan error fixes

### PR DESCRIPTION
**Pull request by dianne-tennent**
This patch fixes some errors detected by a phpstan analysis at
level 0. This doesn't cover the entire directory, only the
files in this patch, as unfortunately I have no more time available.

It addresses errors such as:
 - Access to an undefined property
 - Call to an undefined method

### CODE REVIEW

**Comment by dianne-tennent** [comment]
Just going to redo this with only one commit

File: adodb-memcache.lib.inc.php [hunk 1]
```diff
@@ -401,6 +401,9 @@ public function flushAll($debug=false)
 			return false;
 
 		$del = $this->memcacheLibrary->flush();
+			if ($debug)
+				if (!$del) ADOConnection::outp("flushcache: $filename entry doesn't exist on memcache server!<br>\n");
+				else ADOConnection::outp("flushcache: $filename entry flushed from memcache server!<br>\n");
```
Comment (dregad) [inline adodb-memcache.lib.inc.php:406]: Why add this ? It's the same logic as the test just below, and the $filename variable referenced in the message does not exist.

Also: incorrect indentation level and code does not follow guidelines

File: adodb-pear.inc.php [hunk 2]
```diff
@@ -118,7 +118,7 @@ function factory($type)
 	{
 		include_once(ADODB_DIR."/drivers/adodb-$type.inc.php");
 		$obj = NewADOConnection($type);
-		if (!is_object($obj)) $obj = new PEAR_Error('Unknown Database Driver: '.$dsninfo['phptype'],-1);
+		if (!is_object($obj)) $obj = ADODB_Error_PEAR($type, 'CONNECT', -1, 'Unknown Database Driver: '.$type);
```
Comment (dregad) [inline adodb-pear.inc.php:121]: This is wrong. ADODB_Error_PEAR() does not return a value.

File: adodb-pear.inc.php [hunk 3]
```diff
@@ -164,7 +164,7 @@ function connect($dsn, $options = false)
 
 		@$obj = NewADOConnection($type);
 		if (!is_object($obj)) {
-			$obj = new PEAR_Error('Unknown Database Driver: '.$dsninfo['phptype'],-1);
+			$obj = ADODB_Error_PEAR($type, 'CONNECT', -1, 'Unknown Database Driver: '.$dsninfo['phptype']);
```
Comment (dregad) [inline adodb-pear.inc.php:167]: ditto

File: adodb-xmlschema.inc.php [hunk 4]
```diff
@@ -204,8 +204,13 @@ class dbTable extends dbObject {
...
 	/**
-	 * @var array Field specifier: Meta-information about each field
-	 */
+	* @var object Data object
+	*/
+	var $data;
+
+	/**
+	* @var array Field specifier: Meta-information about each field
+	*/
```
Comment (dregad) [inline adodb-xmlschema.inc.php:213]: OK but please keep 1 space before the `*` on PHPDoc blocks

File: adodb.inc.php [hunk 5]
```diff
@@ -837,7 +842,7 @@ protected function parseHostNameAndPort() {
 	 *
 	 * @return bool
 	 */
-	function Connect($argHostname = "", $argUsername = "", $argPassword = "", $argDatabaseName = "", $forceNew = false) {
+	function _connect($argHostname = "", $argUsername = "", $argPassword = "", $argDatabaseName = "", $forceNew = false) {
```
Comment (dregad) [inline adodb.inc.php:845]: You can't do that ! 

connect() is ADOdb's main connection public method, which relies on driver-specific, protected _connect().

All changes to connect(), nconnect(), pconnect() must be undone.

File: adodb.inc.php [hunk 6]
```diff
@@ -1519,7 +1510,7 @@ function _Execute($sql,$inputarr=false) {
 			}
 			$this->_queryID = _adodb_debug_execute($this, $sql,$inputarr);
 		} else {
-			$this->_queryID = @$this->_query($sql,$inputarr);
+			$this->_queryID = @$this->Query($sql,$inputarr);
```
Comment (dregad) [inline adodb.inc.php:1513]: Same comment as for connect() - query() is the public interface function, _query() is the driver-specific

File: adodb.inc.php [hunk 7]
```diff
@@ -3925,6 +3918,36 @@ function __toString() {
...
+			}
+			return false;
+		}
+		if ($this->_skiprow1) {
+			$pos += 1;
+		}
+		$this->fields = $this->_array[$pos];
+		return true;
+	}
+
```
Comment (dregad) [inline adodb.inc.php:3950]: This introduces a syntax error 
> PHP Fatal error:  Cannot redeclare ADORecordSet::_initRS() in adodb.inc.php on line 3974

Not sure why you are redefining _initRS() and _fetch() methods... And where is this code coming from ?

**Review by dregad — CHANGES_REQUESTED** [review]
This can't be merged in its current state.

PHPDoc blocks must be formatted with a space before the `*`, i.e.
  ```
  /**
  * BAD
  */

  /**
   * CORRECT
   */
  ```

**Comment by dregad** [comment]
Since you did not respond to my review, I'm going to close this without merging; feel free to reopen if you wish to reply to my comments or update the PR.

**Comment by dianne-tennent** [comment]
Thanks for the code review @dregad - and my apologies for the errors. I obviously wasn't testing properly at our end. Happy to leave this closed. I have a smaller patch that we are hoping to include in our repo, so will make another pull request with that one if that's okay. Cheers, Dianne

**Comment by dregad** [comment]
> smaller patch that we are hoping to include in our repo, so will make another pull request with that one if that's okay

By all means do submit, your contributions are welcome!

**Pull request closed without merging**